### ISSUE TITLE
cAdvisor-less, CRI-full Container and Pod Stats

### ISSUE THREAD

**Issue by zvonkok** [kata-containers/kata-containers#10687]
We need to make sure we support the complete list of CRI ListPodSandboxMetrics. At first glimpse I see we do not have any filesystem metrics available in Kata. 

```
// ContainerStats describes a container stats.
type ContainerStats struct {
	CgroupStats  *CgroupStats
	NetworkStats []*NetworkStats
}
```

The idea is to be on par what cAdvisor and ListPodSandboxMetrics will expose to the kubelet. 

References: KEP - https://github.com/kubernetes/enhancements/issues/2371, containerd - https://github.com/containerd/containerd/pull/10691/files

Full list of metric from cAdvisor: https://github.com/google/cadvisor/blob/master/docs/storage/prometheus.md

**Comment by zvonkok** [comment]
@lifupan @Apokleos PTAL regarding runtime-rs.

**Comment by zvonkok** [comment]
For simplification, we could drop cgroupV1 support.

**Comment by zvonkok** [comment]
TODO: Toggle metrics for CC

**Comment by fidencio** [comment]
> For simplification, we could drop cgroupV1 support.

Not sure if this is easily done at this point.  MSFT is using cgroupv1 on their mariner build, for instance.
This requires a deeper conversation.

**Comment by fidencio** [comment]
cc @danmihai1

**Comment by danmihai1** [comment]
> > For simplification, we could drop cgroupV1 support.
> 
> Not sure if this is easily done at this point. MSFT is using cgroupv1 on their mariner build, for instance. This requires a deeper conversation.

@ms-mahuber , @sprt , when do you think we might be able to change the upstream Mariner images from cgroups v1 to v2?

@fidencio , we made this change downstream, so I wouldn't expect this upstream change to take us very long.

Thanks!

**Comment by zvonkok** [comment]
@danmihai1 Is the host also running cgroupv1?

**Comment by danmihai1** [comment]
> @danmihai1 Is the host also running cgroupv1?

@zvonkok , yes. And it turns out that @sprt is out on vacation until next week, so he cannot help yet and this change might take us longer than I was hoping. Sorry.

**Comment by fidencio** [comment]
I've opened https://github.com/kata-containers/kata-containers/pull/10720, which should give us an idea of what will break.

**Comment by zvonkok** [comment]
CRI-O reference implementation for CRI stats https://github.com/cri-o/cri-o/blob/main/internal/lib/stats/metrics.go#L47

**Comment by danmihai1** [comment]
Thanks @fidencio for reminding me today how these settings work!

@fidencio / @zvonkok , 

**I believe the current status is:**

- Mariner Host

Tested mainly using "systemd.legacy_systemd_cgroup_controller=yes systemd.unified_cgroup_hierarchy=0"

- Mariner Guest

Tested mainly using "systemd.legacy_systemd_cgroup_controller=yes systemd.unified_cgroup_hierarchy=0"

- Ubuntu based Guests

Tested mainly using "systemd.unified_cgroup_hierarchy=0" or equivalent build time settings.


**My team's plan for the next round of updates:**

- Mariner Host

systemd built using "default-hierarchy=unified"

- Mariner Guest:

systemd built using "default-hierarchy=unified" and any other cgroups settings that might be preferred by Kata upstream.


**What kind of changes would you like to see?**

- Add to both the Mariner and the Ubuntu Guests "systemd.unified_cgroup_hierarchy=1" during CI?
- Some other changes applicable to Guests?
- Any changes to the Mariner Host?
